docs/plugins.md

```diff
@@ -0,0 +1,115 @@
+
+```js
+<template>
+  <nav-menu-button id="test" icon="loyalty" @nav-click="openNav">Test</nav-menu-button>
+</template>
+ <script>
+'use strict';
+
+// import the navMenuButton component from the global kiln object
+const navMenuButton = window.kiln.utils.components.navMenuButton;
+
+module.exports = {
+  methods: {
+    openNav() {
+      // when the nav-menu-button is clicked, we toggle the drawer open or closed to show or hide the navContent component we set in the index.js file named 'myPlugin'
+      this.$store.dispatch('toggleDrawer', 'myPlugin');
+    }
+  },
+  components: {
+    navMenuButton
+  }
+};
+</script>
+```
+Which will add the Test button to the nav, like this.
+
+![nav button](/clay-kiln/img/navbutton.png)
+
+---
+
+You can also add buttons to the Kiln Toolbar by adding them to the `windows.kiln.toolbarButtons` array.
+```js
+window.kiln.toolbarButtons['test'] = require('./toolbar-button.vue');
+```
+Then the `toolbar-button.vue` file would then look very similar to the `nav-button.vue` file above. The biggest change is that you wouldn't use a nav-menu-button component because those are styled so as to be used in the left nav.
+```js
+<template>
+  <UiIconButton id="test" icon="favorite" @click="openNav" />
+</template>
+ <script>
+'use strict';
+const UiIconButton = window.kiln.utils.components.UiIconButton;
+
+module.exports = {
+  data() {
+    return {};
+  },
+  methods: {
+    openNav() {
+      this.$store.dispatch('toggleDrawer', 'test');
+    }
+  },
+  components: {
+    UiIconButton
+  }
+};
+</script>
+```
+
+This will add a button that looks like this
+
+![Toolbar button](/clay-kiln/img/toolbarbutton.png)
+
+---
+
+## Plugin Content
+
+Content for you plugin will appear within the nav content drawer and should be added to the `window.kiln.navcontent` array.
+
+```js
+  window.kiln.navContent['myPlugin'] = require('./main.vue');
+```
+
+Plugins are most often built as a Vue App that exists inside the Kiln Vue App and can be as simple or as complex as you need.  As a part of the Kiln app it has access to the Vuex Store and all of the data it contains and can both read and write to that data, having access to all Kiln's Vuex Actions.
+
+Plugins can range from something as simple as this
+
+![Clear Cache Plugin](/clay-kiln/img/clear_cache.png)
+
+to something as complex as this
+
+![Agora Plugin](/clay-kiln/img/agora.png)
+
+In fact, a plugin can be even simpler than either of those and doesn't actually require anything to be added to the `window.kiln.navcontent` array at all. Instead of opening a content drawer, clicking the nav button could trigger a JavaScript function that makes an API call to perform some backend function, or it could update some piece of data in Kiln's state, or any other JavaScript function that you write.
```

Comment:
```suggestion
In fact, a plugin can be even simpler than either of those and doesn't actually require anything to be added to the `window.kiln.navContent` array at all. Instead of opening a content drawer, clicking the nav button could trigger a JavaScript function that makes an API call to perform some backend function, or it could update some piece of data in Kiln's state, or any other JavaScript function that you write.
```